hop/configure.py

```diff
@@ -51,6 +52,7 @@ def write_config_file(config_file, username, password):
 
     os.makedirs(os.path.dirname(config_file), exist_ok=True)
     with open(config_file, "w") as f:
+        os.chmod(config_file, stat.S_IRUSR | stat.S_IWUSR)
```

Comment:
It sure feels like this should happen outside the `with` body. I suppose this is fine, but it raised my eyebrows at least.

Reply:
I think it can't happen earlier, as the file is not gauranteed to exist until we open it. I suppose this does leave a timing window in which someone else could open the file before we read protect it and then read the data we subsequently write, but I'm not sure there's a way to entirely prevent that.

Reply:
Oh, I meant I'd expect it to be done after the file has been closed, at the end of the with block.

It's certainly possible to do this all-in-one-go by passing a file descriptor to `open`. You can make the fd with `os.open`, which can take the permission bits. It's just a bit hairier in general but it's possible.

Reply:
Right, I forgot `open(2)` could do that; it's much cleaner.